src/cursor/abstract_cursor.ts

```diff
@@ -481,6 +495,7 @@ export abstract class AbstractCursor<
         }
 
         yield document;
+        throwIfAborted(this.signal);
```

Comment:
Should a line like this also be added before the `await` above?

Reply:
Yes, unless I'm mistaken I think generally we _must_ always check the state before and after an await. I'm hoping to enumerate the possible state change (before we enter an await and after we resolve) in the tests in such a way that they will fail if I omit it 🤞🏻